include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -281,7 +281,10 @@ struct take_view_simple
     _R __r;
     _Size __n;
 
-    take_view_simple(_R __rng, _Size __size) : __r(__rng), __n(__size) { assert(__n >= 0 && __n < __r.size()); }
+    take_view_simple(_R __rng, _Size __size) : __r(__rng), __n(::std::min(__size, _Size(__r.size())))
+    {
+        assert(__n >= 0);
+    }
```

Comment:
Handling it like this makes `size()` faster (not recalculating `std::min` for every `size()` call), but loses a bit of information as `__n` is public and may not match the input `__size` when it exceeds the input range.  

I still think this is correct, but I'm mentioning it here.  Perhaps it is worth a comment above `__n` in case anyone is expecting different semantics (like `__n` to be always equal to the input `__size`).

Reply:
`__n` is public in the C++ sense but not supposed to be used directly, i.e. it's an undocumented implementation detail.

However, I am not sure if such an implementation is correct. We assume that the size of the underlying range is fixed, i.e. does not change during the lifetime of the view. But is it a correct assumption?

Reply:
@akukanov I have taken a look and you are correct. If the underlying range is modified (ex: gets smaller), then it should still be safe. I will make the changes to ensure this.

Reply:
The question is, however, whether the underlying range **can** get bigger or smaller. As far as I understand, these views are for internal use and are intentionally simple, avoiding overheads where generality is not needed. So maybe the assumption about the size is correct, and all that we need to do is document it in the code.

Reply:
If this is the direction we would want to take,  I think I may prefer to go back to limiting it via assert to be within `0<=n<=size`.  This seems a safer way to limit generality in favor of performance.

Reply:
I agree with both points. I think just modifying the initial assert for correctness in `take_view_simple` and `drop_view_simple` will be the best way to go.